### PR TITLE
Add input validation to consultation form

### PR DESCRIPTION
**Pull request by Ovifer13**
**Issue #161**

### CODE REVIEW

File: pages/consultation/index.tsx [hunk 1]
```diff
@@ -39,10 +39,11 @@ const ConsultationPage: NextPage = () => {
             },
             method: 'POST'
         })
-        router.push('/consultation/thanks')
         const { error } = await res.json()
         if (error) {
             console.log(error)
+        } else {
+            router.push('/consultation/thanks')
```
Comment (sofiaromorales) [inline pages/consultation/index.tsx:46]: Use `try/catch`

File: components/ContactInformation.tsx [hunk 2]
```diff
@@ -35,44 +45,81 @@ const ContactInformation = ({
...
+                            placeholder={input.name}
+                            required
+                            value={input.value}
+                            onChange={(e) => input.onChange(e.target.value)}
+                        />
+                        <span className='absolute text-xs px-4 font-medium text-gray-500 transition-all left-3 peer-focus:text-xs peer-focus:top-3 peer-focus:translate-y-0 peer-placeholder-shown:top-1/2 peer-placeholder-shown:-translate-y-1/2 peer-placeholder-shown:text-sm'>
+                            { input.name }
+                        </span>
+                    </label>
+                    { input.error && <span className='px-5 text-red-600'>{ INVALID } { input.name }</span> }
```
Comment (sofiaromorales) [inline components/ContactInformation.tsx:111]: Fix format

File: components/ProjectGoalsForm.tsx [hunk 3]
```diff
@@ -89,25 +101,33 @@ const ProjectGoalsForm = ({
                     {WHICH_SERVICE_BEST_FITS}
                 </Paragraph>
                 { renderServiceTypes() }
+                { serviceTypeError && <span className='px-5 text-red-600'>{ PLEASE_SELECT_AT_LEAST_ONE_SERVICE }</span> }
```
Comment (sofiaromorales) [inline components/ProjectGoalsForm.tsx:104]: Fix format

File: pages/consultation/index.tsx [hunk 4]
```diff
@@ -66,24 +81,27 @@ const ConsultationPage: NextPage = () => {
...
+                    setServiceInformationError={setServiceInformationError}
                 />
             </FormSection>
             <Section>
                 <div onClick={(e: any) => handleSubmit(e)}>
-                    <Button variant='tertiary'>
+                    <Button variant='tertiary' disabled={disabledButton}>
                         {SUBMIT}
                     </Button>
                 </div>
```
Comment (sofiaromorales) [inline pages/consultation/index.tsx:107]: Submit is still getting called even when the form is incorrect 🥴

File: utilities/validations/regex.ts [hunk 5]
```diff
@@ -0,0 +1,2 @@
+export const validEmail: RegExp = /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i;
+export const validNumber: RegExp = /^\d+$/;
```
Comment (sofiaromorales) [inline utilities/validations/regex.ts:2]: Change to `utils/validations.tsx`

**Review by sofiaromorales — CHANGES_REQUESTED** [review]

**Comment by sofiaromorales** [comment]
@Ovifer13 pls take a look into [d7f28ca](https://github.com/setlife-network/setlife-solutions/pull/177/commits/d7f28cadb995611d571bc68f760d6e20bb1a6f49) and have it present for upcoming pr's

File: pages/consultation/index.tsx (hunk 4 above)
Comment (Ovifer13) [reply pages/consultation/index.tsx:107]: Yup, that's because of the button comoponent, see that he onClick is in the div and not in the button

Comment (sofiaromorales) [reply pages/consultation/index.tsx:107]: Ik, but it shouldn't get sent, pls fix

**Comment by sofiaromorales** [comment]
@Ovifer13 all the contact information is required, rn I'm able to send the form only by filling the name field (+ the other sections of the form), pls fix
Screen Shot 2022-09-08 at 2 35 36 PM

**Review by sofiaromorales — APPROVED** [review]

**Pull request merged**